### DISCUSSION TITLE
GitHub Codespaces: how to set x86_64, AMD64, ARM64 platform?

### DISCUSSION THREAD

**Discussion by ultraGentle** [community/community#8507] [Codespaces] [answered]
First, the question: **is there a way to choose the architecture (e.g. x86_64, AMD64, ARM64) for a GitHub Codespace?** 

Here's what I've found so far: 

**Attempt 1 (not working):**

From within GitHub.com, you can choose the "machine" for a Codespace, but the only options are RAM and disk size.  

**Attempt 2 (in progress; input appreciated):**

This route seems more promising, but it's all new to me (containerization, codespaces, docker).  

When you create a Codespace, you can specify options by creating a top-level `.devcontainer` folder with two files: `devcontainer.json` and `Dockerfile`  

Here you can customize runtimes, installed packages, etc., but the docs don't say anything about determining architecture...

...however, the VSCode docs for `devcontainer.json` has a `runArgs` option, which "accepts Docker CLI arguments"...  

and the Docker CLI docs on --platform say you should be able to pass `--platform linux/amd64` or `--platform linux/arm64`, but...

**1. Does GitHub Codespaces support this?**   
**2. Would you pass it in the Dockerfile or devcontainer.json -- does it not matter?**  
3. For windows/x86_64, presumably you'd need a license?    

Thanks for any suggestions!

Note, this is all from my Stack Overflow question, which didn't get an answer, hence asking here -- it would be great if the devs would monitor Codespaces questions there, too!

References:   
https://code.visualstudio.com/docs/remote/devcontainerjson-reference  
https://docs.docker.com/engine/reference/commandline/run/    
https://docs.docker.com/engine/reference/builder/  
https://docs.docker.com/desktop/multi-arch/  
https://docs.docker.com/buildx/working-with-buildx/

**Comment by asciimike** [comment, marked as answer]
Currently Codespaces only offers Linux as the OS and x86_64 as the architecture for the host machines. Other OSes and architectures are under consideration, and would likely come from Azure's pool of VM types: https://azure.microsoft.com/en-us/pricing/details/virtual-machines/series/

There's likely a world where you can run docker-in-docker (seems like what you're doing?) ala https://github.com/microsoft/vscode-dev-containers/blob/main/script-library/docs/docker-in-docker.md and doing something like what's suggested in https://matchboxdorry.gitbooks.io/matchboxblog/content/blogs/build_and_run_arm_images.html

**Reply by btwiuse** [reply]
Now github offers arm64 action runners for free. Does it mean we'll have it for codespaces in the near future too?

**Reply by Kreijstal** [reply]
I would love for arm64 codespaces
